fe/fe-core/src/main/java/org/apache/doris/statistics/StatisticsJob.java

```diff
@@ -18,62 +18,200 @@
+     * to be executed tasks.
+     */
+    private List<StatisticsTask> tasks = Lists.newArrayList();
+
     private JobState jobState = JobState.PENDING;
-    // optional
-    // to be collected table stats
-    private List<Long> tableId = Lists.newArrayList();
-    // to be collected column stats
-    private Map<Long, List<String>> tableIdToColumnName = Maps.newHashMap();
-    private Map<String, String> properties;
-    // end
+    private final List<String> errorMsgs = Lists.newArrayList();
 
-    private List<StatisticsTask> taskList = Lists.newArrayList();
+    private final long createTime = System.currentTimeMillis();
+    private long startTime = -1L;
+    private long finishTime = -1L;
+    private int progress = 0;
+
+    public StatisticsJob(Long dbId,
+                         Set<Long> tblIds,
+                         Map<Long, List<String>> tableIdToColumnName) {
+        this.dbId = dbId;
+        this.tblIds = tblIds;
+        this.tableIdToColumnName = tableIdToColumnName;
+    }
 
     public long getId() {
-        return id;
+        return this.id;
+    }
+
+    public void setId(long id) {
+        this.id = id;
+    }
+
+    public long getDbId() {
+        return this.dbId;
+    }
+
+    public Set<Long> getTblIds() {
+        return this.tblIds;
+    }
+
+    public Map<Long, List<String>> getTableIdToColumnName() {
+        return this.tableIdToColumnName;
+    }
+
+    public long getTaskTimeout() {
+        return taskTimeout;
     }
 
-    /*
-        AnalyzeStmt: Analyze t1(c1), t2
-        StatisticsJob:
-          tableId [t1, t2]
-          tableIdToColumnName <t1, [c1]> <t2, [c1,c2,c3]>
-         */
-    public static StatisticsJob fromAnalyzeStmt(AnalyzeStmt analyzeStmt) {
-        // TODO
-        return new StatisticsJob();
+    public List<StatisticsTask> getTasks() {
+        return this.tasks;
+    }
+
+    public void setTasks(List<StatisticsTask> tasks) {
+        this.tasks = tasks;
+    }
+
+    public List<String> getErrorMsgs() {
+        return errorMsgs;
+    }
+
+    public JobState getJobState() {
+        return this.jobState;
+    }
+
+    public long getCreateTime() {
+        return this.createTime;
+    }
+
+    public long getStartTime() {
+        return this.startTime;
+    }
+
+    public void setStartTime(long startTime) {
+        this.startTime = startTime;
+    }
+
+    public long getFinishTime() {
+        return this.finishTime;
+    }
+
+    public void setFinishTime(long finishTime) {
+        this.finishTime = finishTime;
+    }
+
+    public int getProgress() {
+        return this.progress;
+    }
+
+    public void setProgress(int progress) {
+        this.progress = progress;
+    }
+
+    private void setOptional(AnalyzeStmt stmt) {
+        if (stmt.getTaskTimeout() != -1) {
+            this.taskTimeout = stmt.getTaskTimeout();
+        }
+    }
+
+    public synchronized void updateJobState(JobState jobState) {
+        // PENDING -> SCHEDULING/FAILED/CANCELLED
```

Comment:
If there is a invalid state switch, it will thrown exception

Reply:
ok

Reply:
It may be better to log instead of exceptions. For example, if multiple tasks fail, then there may be job status updates from FAILED to FAILED. This is actually a normal behavior (we don’t need to do anything), adding Exceptions may make the code complicated~